### DISCUSSION TITLE
Unable to import Ssh-Key with Passphrase

### DISCUSSION THREAD

**Discussion by spammads** [dani-garcia/vaultwarden#5331] [Q&A]
### Vaultwarden Support String

asdf

### Vaultwarden Build Version

latest

### Deployment method

Official Container Image

### Reverse Proxy

latest Nginx

### Host/Server Operating System

Linux

### Operating System Version

Mac Os latest

### Clients

Desktop

### Client Version

latest

### Steps To Reproduce

1. Add Item
2. Choose SSH Key
3. Import key from clipboard
4. Enter Passphrase
5. Get Error-Message: "SshKey is invalid"

I tested the import with a standard-ssh-key (rsa, generated with `ssh-keygen -t rsa -b 4096 -C "your_email@example.com"`) and this works

### Expected Result

Import the key

### Actual Result

Error Message that the key is invalid

### Screenshots or Videos

<img width="465" alt="Screenshot 2024-12-29 at 15 51 17" src="https://github.com/user-attachments/assets/c9134929-5534-4d7c-bee9-afd8576f7a5a" />
<img width="694" alt="Screenshot 2024-12-29 at 15 51 23" src="https://github.com/user-attachments/assets/afa15b9f-b16b-40df-b08a-30efd1f3df7f" />

**Comment by obstschale** [comment]
@spammads Did you find a solution? I have the same problems and I cannot figure out how to import my keys

**Reply by spammads** [reply]
Nope. I migrated to another password-provider (but not bc of this issue 🫣)

**Comment by strasharo** [comment]
Same here for me with both ed25519 and RSA keys. Has anyone found a workaround for this?

**Comment by bokkabonga** [comment]
Bit of a stretch here. But from https://bitwarden.com/help/ssh-agent/ "The SSH Agent requires release version 2025.1.2 or newer."

The current Webvault in Vaultwarden is "2025.1.1", so that might explain the problem. But even when this project updates to 2025.1.2 (or later) I´m not sure if the ssh-agent will be supported as it might be locked behind a different license...

**Comment by Peronia** [comment]
The problem also exists with vaultwarden version 1.34.1.

**Reply by Peronia** [reply]
I'm using version 1.34.3 now and still not able to save SSH keys in Vaultwarden.
I tried it via the desktop app on linux and mac. I tried my own password protected key and I copied that one that is generated when I hit + and try to paste it on a new entry. I see only "Ein Fehler ist aufgetreten" and nothing more.
A friend tried the same with underlying official Bitwarden and for him it works fine.
Are you aware of that @BlackDex (or any other, the discussion is quite old)?

**Reply by bokkabonga** [reply]
I just tried and got no problem with this via the desktop client.  How did you generate your key? Does your key meet the following?

At this time, imported SSH keys from Putty are not compatible.

Imported keys must be in OpenSSH or PKCS#8 format.

**Reply by Peronia** [reply]
I created it years ago the default cli way. I used it nearly daily to connect with SSH.
It looks like this:
```
-----BEGIN OPENSSH PRIVATE KEY-----
hereyourkey
-----END OPENSSH PRIVATE KEY-----
```
So I would say this sould be fine. The key uses RSA.
Also - as I wrote - I tried to import that one that is generated from the desktop client itself. Thats also not working with the same error message.

**Reply by Peronia** [reply]
Okay, I stumbled across `EXPERIMENTAL_CLIENT_FEATURE_FLAGS` by chance (https://github.com/dani-garcia/vaultwarden/blob/main/.env.template#L384C37-L384C60). If you set `ssh-agent` and `ssh-key-vault-item` there, keys (RSA and ED25519) can be stored without a password. It still doesn't work with a password.
I haven't found any documentation on this.
Also, the default value `fido2-vault-credentials` doesn't make sense, as this value is not mentioned in the flags.
